### PR TITLE
RR: Initialize subchannel connectivity state

### PR DESCRIPTION
**Pull request by dgquintas**

### CODE REVIEW

File: test/cpp/end2end/client_lb_end2end_test.cc [hunk 1]
```diff
@@ -450,6 +450,43 @@ TEST_F(ClientLbEnd2endTest, RoundRobinUpdates) {
...
+  const int kNumServers = 3;
+  StartServers(kNumServers);
+  ResetStub("round_robin");
+  std::vector<int> ports;
+
+  // Start with a single server.
+  ports.emplace_back(servers_[0]->port_);
+  SetNextResolution(ports);
+  WaitForServer(0);
+  // Send RPCs. They should all go servers_[0]
```
Comment (markdroth) [inline test/cpp/end2end/client_lb_end2end_test.cc]: s/go/go to/

File: test/cpp/end2end/client_lb_end2end_test.cc [hunk 2]
```diff
@@ -450,6 +450,43 @@ TEST_F(ClientLbEnd2endTest, RoundRobinUpdates) {
   EXPECT_EQ("round_robin", channel_->GetLoadBalancingPolicyName());
 }
 
+TEST_F(ClientLbEnd2endTest, RoundRobinUpdateInError) {
+  // Start servers and send one RPC per server.
```
Comment (markdroth) [inline test/cpp/end2end/client_lb_end2end_test.cc]: This comment seems inaccurate.

File: test/cpp/end2end/client_lb_end2end_test.cc [hunk 3]
```diff
@@ -450,6 +450,43 @@ TEST_F(ClientLbEnd2endTest, RoundRobinUpdates) {
...
+  SetNextResolution(ports);
+  WaitForServer(0);
+  // Send RPCs. They should all go servers_[0]
+  for (size_t i = 0; i < 10; ++i) SendRpc();
+  EXPECT_EQ(10, servers_[0]->service_.request_count());
+  EXPECT_EQ(0, servers_[1]->service_.request_count());
+  EXPECT_EQ(0, servers_[2]->service_.request_count());
+  servers_[0]->service_.ResetCounters();
+
+  // All servers, but one is shutdown.
```
Comment (markdroth) [inline test/cpp/end2end/client_lb_end2end_test.cc]: s/All servers/Add remaining servers/

File: test/cpp/end2end/client_lb_end2end_test.cc [hunk 4]
```diff
@@ -450,6 +450,43 @@ TEST_F(ClientLbEnd2endTest, RoundRobinUpdates) {
...
+  servers_[1]->Shutdown(false);
+  ports.clear();
```
Comment (markdroth) [inline test/cpp/end2end/client_lb_end2end_test.cc]: Can remove this line and the next -- all we need to do here is to add the last two ports.

File: test/cpp/end2end/client_lb_end2end_test.cc [hunk 5]
```diff
@@ -450,6 +450,43 @@ TEST_F(ClientLbEnd2endTest, RoundRobinUpdates) {
...
+  WaitForServer(0);
+  WaitForServer(2);
+
+  // Send three RPCs, one per server.
+  for (size_t i = 0; i < kNumServers; ++i) SendRpc();
+  // The server in shutdown shouldn't receive any.
+  EXPECT_EQ(0, servers_[1]->service_.request_count());
+
+  // Check LB policy name for the channel.
+  EXPECT_EQ("round_robin", channel_->GetLoadBalancingPolicyName());
```
Comment (markdroth) [inline test/cpp/end2end/client_lb_end2end_test.cc]: We don't really need to do this in every test.  We just need to test it once, to prove that requesting a particular policy does the right thing.

File: test/cpp/end2end/client_lb_end2end_test.cc (hunk 2 above)
Comment (dgquintas) [inline test/cpp/end2end/client_lb_end2end_test.cc]: Removed

File: test/cpp/end2end/client_lb_end2end_test.cc (hunk 1 above)
Comment (dgquintas) [inline test/cpp/end2end/client_lb_end2end_test.cc]: Done

File: test/cpp/end2end/client_lb_end2end_test.cc (hunk 3 above)
Comment (dgquintas) [inline test/cpp/end2end/client_lb_end2end_test.cc]: Rephrased

File: test/cpp/end2end/client_lb_end2end_test.cc (hunk 4 above)
Comment (dgquintas) [inline test/cpp/end2end/client_lb_end2end_test.cc]: Done

File: test/cpp/end2end/client_lb_end2end_test.cc (hunk 5 above)
Comment (dgquintas) [inline test/cpp/end2end/client_lb_end2end_test.cc]: Done

**Pull request merged**